outward_assembly/pipeline_steps.py

```diff
@@ -107,13 +124,176 @@ def _assemble_contigs(workdir: PathLike, iter: int, freq_filter: bool) -> None:
             subprocess.run(cmd, stdout=log, stderr=log, check=True)
 
 
+def _contig_ids_by_seed_ahocorasick(
+    records: List[SeqRecord],
+    seed_seqs: List[Seq],
+) -> Dict[int, SeqOrientation]:
+    """
+    Find which contigs contain which seeds using Aho-Corasick algorithm.
+
+    Args:
+        records: List of SeqRecord objects (contigs)
+        seed_seqs: List of seed sequences to search for
+
+    Returns:
+        Dict mapping contig index to orientation (FORWARD or REVERSE)
+    """
+
+    if not seed_seqs or not records:
+        return {}
+
+    automaton = ahocorasick.Automaton()
+
+    for seed_idx, seed in enumerate(seed_seqs):
+        seed_str = str(seed).upper()
+        seed_rc = _reverse_complement(seed_str)
+
+        # Forward seed: if this matches, contig is FORWARD relative to seed
+        automaton.add_word(seed_str, (seed_idx, SeqOrientation.FORWARD))
+
+        # Reverse complement: if this matches, contig is REVERSE relative to seed
+        if seed_rc != seed_str:  # avoid palindrome duplicates
+            automaton.add_word(seed_rc, (seed_idx, SeqOrientation.REVERSE))
+
+    automaton.make_automaton()
+
+    matches: Dict[int, SeqOrientation] = {}
+
+    for contig_idx, record in enumerate(records):
+        contig_seq = str(record.seq).upper()
+
+        for _end_pos, (_seed_idx, orientation) in automaton.iter(contig_seq):
+            matches[contig_idx] = orientation
+            break 
+
+    return matches
+
+
+def _contig_ids_by_seed_bowtie2(
```

Comment:
I vaguely recall from our prior conversation that the aho corasick approach dominated the bowtie2 approach. Is that right? When should we use bowtie2 instead of AC?